### DISCUSSION TITLE
Best Practices for Securely Submitting Changes in a Private Project on GitHub

### DISCUSSION THREAD

**Discussion by dariohsb** [community/community#86586] [Code Security] [answered]
### Select Topic Area

General

### Body

I'm currently working on a private project and am looking for advice on the best practices for securely submitting changes. My current approach involves working on a portion of the project, making a commit, and then repeating this process until completion. However, I'm concerned about maintaining a secure and efficient workflow, especially in scenarios where I might need to revert changes.

Is there a standard or recommended method for submitting changes to a private project on GitHub that ensures safety and allows for easy reversion if needed? I'm keen on understanding if there's a more effective strategy than my current method of continuous commits.

Any insights or recommendations would be greatly appreciated. Thank you!

**Comment by faridnec** [comment, marked as answer]
you can create a new branch for each feature or bug fix. This keeps your changes isolated and makes it easier to revert or merge later.
Use a consistent naming convention for branches, such as feature/..., bugfix/..., or a similar structure. Also, make frequent, small commits with clear and concise messages. Each commit should represent a logical unit of change and avoid committing large, unrelated changes together, as it can make reverting more challenging.

This [article](https://medium.com/forcodesake/mastering-the-git-workflow-github-stage-area-commit-history-repository-6c8d1a6ac8f3) might be helpful for further detail
